polytracker/src/dfsan_pass/BBSplittingPass.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * BBSplittingPass.cpp
+ *
+ *  Created on: Aug 14, 2020
+ *      Author: Evan Sultanik, Trail of Bits
+ */
+
+#include <iostream>
+#include <string>
+
+#include "llvm/IR/InstrTypes.h"
+#include "llvm/IR/Instructions.h"
+
+#include "polytracker/bb_splitting_pass.h"
+
+using llvm::BasicBlock;
+using llvm::BranchInst;
+using llvm::CallInst;
+using llvm::dyn_cast;
+using llvm::Function;
+using llvm::Instruction;
+using llvm::isa;
+using llvm::TerminatorInst;
+
+namespace polytracker {
+
+bool BBSplittingPass::analyzeBasicBlock(BasicBlock &basicBlock) const {
+  bool modified = false;
+
+  for (Instruction *inst = &basicBlock.front();
+       inst && !isa<TerminatorInst>(inst); inst = inst->getNextNode()) {
+    if (auto call = dyn_cast<CallInst>(inst)) {
+      // Is the call immediately followed by an unconditional branch?
+      // if so, that's the only case that is okay:
+      Instruction *next = inst->getNextNode();
+      if (next == nullptr) {
+        continue;
+      } else if (auto branch = dyn_cast<BranchInst>(next)) {
+        if (branch->isUnconditional()) {
+          // The next instruction after the call is an unconditional branch,
+          // so it's okay to leave it.
+          continue;
+        }
+      }
+      std::string fname;
+      if (auto function = call->getCalledFunction()) {
+        if (function->hasName()) {
+          fname = function->getName().data();
+        }
+      } else if (auto v = call->getCalledValue()->stripPointerCasts()) {
+        if (v->hasName()) {
+          fname = v->getName().data();
+        }
+      }
+      // We need to split this BB into a new one after the call
+      modified = true;
```

Comment:
I don't see if this variable is used anywhere else besides the return, but I might be blind. Maybe we can just return True for this branch where we split, and false for the other one (which you already do)

Reply:
So, uh, I apologize, but it looks like I fixed this code in the `mimid` branch and didn't push it back to this branch 😫 Sorry for the confusion, and thank you for reviewing this, but I think I should just close this PR and we should focus on the version in PR #93